File: basic-proxy-impl/src/test/java/com/test/cases/ArrayParameterMethodTest.java
package com.test.cases;

import java.util.List;

@For(ArrayParameterMethod.class)
public class ArrayParameterMethodTest extends TestBase {
    public ArrayParameterMethodTest() throws Exception {
        super();
    }
}

interface ArrayParameterMethod {
    void f(int[] p1, long[] p2, float[] p3, double[] p4, int[][][][] p5, Object[][][] p6, Void[][][] p7, List[][] p9);
}

File: basic-proxy-kotlin-impl/src/test/kotlin/com/test/cases/ArrayParameterMethodTest.kt
package com.test.cases

@For(ArrayParameterMethod::class)
class ArrayParameterMethodTest : TestBase()

interface ArrayParameterMethod {
    fun f(
        p1: Array<Int>,
        p2: Array<Long>,
        p3: Array<Float>,
        p4: Array<Double>,
        p5: Array<Array<Array<Array<Int>>>>,
        p6: Array<Array<Array<Any>>>,
        p7: Array<Array<Array<Unit>>>,
        p9: Array<Array<List<Any>>>
    )
}